blockchain/blockchain.go

```diff
@@ -70,34 +42,61 @@ func NewBlockchain(database db.DB, network utils.Network) *Blockchain {
 }
 
 // Height returns the latest block height. If blockchain is empty nil is returned.
-func (b *Blockchain) Height() *uint64 {
-	headBlock, err := b.Head()
-	if err != nil && !errors.Is(err, db.ErrKeyNotFound) {
-		panic(fmt.Sprintf("failed to retrieved head block: %v", err))
-	}
-	if headBlock != nil {
-		return &headBlock.Number
+func (b *Blockchain) Height() (uint64, error) {
+	var height uint64
+
+	return height, b.database.View(func(txn db.Transaction) error {
+		dbHeight, err := b.height(txn)
+		height = dbHeight
+		return err
+	})
+}
+
+func (b *Blockchain) height(txn db.Transaction) (uint64, error) {
+	heightBin, err := txn.Get(db.ChainHeight.Key())
```

Comment:
We were duplicating the head block by storing it whole before this PR. We shouldn't duplicate any data, but I am open to any other suggestions. The way we store blocks will probably change a lot when we change the block structure to include transactions and receipts anyways. So let's just keep it for another PR?

Reply:
I was ok with a single blocks duplication but I agree no duplication is better.  A better approach can be to append the hash to the block number (`ChainHeight -> blockNumber+blockHash`) this would then uniquely identify a single block. Let's keep this in mind for the future when we implement reorg.